serverless.yaml

```diff
@@ -44,7 +48,10 @@ provider:
     FHIR_BINARY_BUCKET: !Ref FHIRBinaryBucket
     ELASTICSEARCH_DOMAIN_ENDPOINT: !Join ['', ['https://', !GetAtt ElasticSearchDomain.DomainEndpoint]]
     OAUTH2_DOMAIN_ENDPOINT:
-      !Join ['', ['https://', !Ref UserPoolDomain, !Sub '.auth.${AWS::Region}.amazoncognito.com/oauth2']]
+      !If 
+        - isUsingDefaultUserPool
+        - !Join ['', ['https://', !Ref UserPoolDomain, !Sub '.auth.${AWS::Region}.amazoncognito.com/oauth2']]
+        - !Join ['', ['https://', !Ref ExtUserPoolDomain, !Sub '.auth.${AWS::Region}.amazoncognito.com/oauth2']]

```

Comment:
Can the `!If` be where `!Ref UserPoolDomain` is? That would prevent a bit of code duplication.